cpp/src/codegen/CMakeLists.txt

```diff
@@ -84,18 +87,39 @@ install(
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
 
+# Pre-compiled .so library for function helpers.
+add_library(gandiva_helpers SHARED
+  ${SHARED_HELPER_FILES}
+  function_holder_stubs.cc)
+
+target_compile_definitions(gandiva_helpers
+  PRIVATE -DGDV_HELPERS
+)
+
+target_include_directories(gandiva_helpers
+  PRIVATE
+    ${CMAKE_SOURCE_DIR}/include
+    ${CMAKE_SOURCE_DIR}/src
+    $<TARGET_PROPERTY:ARROW::ARROW_SHARED,INTERFACE_INCLUDE_DIRECTORIES>
```

Comment:
do the helpers need the arrow includes?

Reply:
yes - the LikeHolder::Make takes FunctionNode as param, and node.h includes arrow files (for data type).